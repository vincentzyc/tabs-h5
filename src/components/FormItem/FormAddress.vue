<template>
  <div class="flex align-middle form-item border-bnone">
    <div class="form-label">详细地址</div>
    <div class="form-input flex-auto flex align-middle">
      <input
        @blur="checkAddress(model)"
        :maxlength="50"
        placeholder="街道/镇+村/小区/楼+门牌号"
        type="text"
        v-model.trim="model"
      />
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue';
import { checkAddress } from '@/composition/business/use-verify-data'


export default defineComponent({
  name: "FormAddress",
  props: {
    modelValue: {
      required: true,
      type: String,
    }
  },
  emits: ['update:modelValue'],
  setup(props, { emit }) {
    const model = computed<string>({
      get() {
        return props.modelValue;
      },
      set(val) {
        emit("update:modelValue", val);
      }
    })
    return {
      model,
      checkAddress,
    }
  }
})
</script>