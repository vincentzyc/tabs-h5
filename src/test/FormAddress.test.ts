import { mount } from '@vue/test-utils'
import FormAddress from '@/components/FormItem/FormAddress.vue'

test('mount FormAddress component', async () => {
  expect(FormAddress).toBeTruthy()

  const wrapper = mount(FormAddress, {
    props: {
      modelValue: 'initialText',
      'onUpdate:modelValue': (e: string) => wrapper.setProps({ modelValue: e })
    }
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('input').setValue('测试详细地址')

  expect(wrapper.props('modelValue')).toBe('测试详细地址')

  const emitEvent = wrapper.emitted('update:modelValue')

  expect(emitEvent).toHaveLength(1)
})