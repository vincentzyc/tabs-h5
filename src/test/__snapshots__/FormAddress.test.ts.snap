// Vitest Snapshot v1

exports[`mount FormAddress component 1`] = `
"<div class=\\"flex align-middle form-item border-bnone\\">
  <div class=\\"form-label\\">详细地址</div>
  <div class=\\"form-input flex-auto flex align-middle\\"><input maxlength=\\"50\\" placeholder=\\"街道/镇+村/小区/楼+门牌号\\" type=\\"text\\"></div>
</div>"
`;
